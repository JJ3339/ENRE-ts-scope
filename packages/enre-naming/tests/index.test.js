import ENREName from '../src/name';

describe('ENREName', () => {
    it('of valid string identifier', () => {
        expect(new ENREName('Norm', 'foo')).toMatchSnapshot();

        expect(ENREName.fromString('foo')).toMatchSnapshot();
    });

    it('of package', () => {
        expect(new ENREName('Pkg', 'foo')).toMatchSnapshot();

        expect(ENREName.fromString('<Pkg foo>')).toMatchSnapshot();
    });

    it('of file name', () => {
        expect(new ENREName('File', 'foo.js')).toMatchSnapshot();

        expect(ENREName.fromString('<File foo.js>')).toMatchSnapshot();
    });

    describe('of anonymous', () => {
        it('Function', () => {
            expect(new ENREName('Anon', 'Function')).toMatchSnapshot();

            expect(ENREName.fromString('<Anon Function>')).toMatchSnapshot();
        });

        it('ArrowFunction', () => {
            expect(new ENREName('Anon', 'ArrowFunction')).toMatchSnapshot();

            expect(ENREName.fromString('<Anon ArrowFunction>')).toMatchSnapshot();
        });

        it('Class', () => {
            expect(new ENREName('Anon', 'Class')).toMatchSnapshot();

            expect(ENREName.fromString('<Anon Class>')).toMatchSnapshot();
        });

        it('Block', () => {
            expect(new ENREName('Anon', 'Block')).toMatchSnapshot();

            expect(ENREName.fromString('<Anon Block>')).toMatchSnapshot();

            expect(new ENREName('Anon', 'Block', '1:1')).toMatchSnapshot();

            expect(ENREName.fromString('<Block 1:1>')).toMatchSnapshot();
        });
    });

    describe('of signature declaration', () => {
        it('Callable', () => {
            expect(new ENREName('Sig', 'Callable')).toMatchSnapshot();

            expect(ENREName.fromString('<Sig Callable>')).toMatchSnapshot();
        });

        it('NumberIndex', () => {
            expect(new ENREName('Sig', 'NumberIndex')).toMatchSnapshot();

            expect(ENREName.fromString('<Sig NumberIndex>')).toMatchSnapshot();
        });

        it('StringIndex', () => {
            expect(new ENREName('Sig', 'StringIndex')).toMatchSnapshot();

            expect(ENREName.fromString('<Sig StringIndex>')).toMatchSnapshot();
        });
    });

    it('of string literal', () => {
        expect(new ENREName('Str', 'a-not-valid-identifier')).toMatchSnapshot();

        expect(ENREName.fromString('<Str a-not-valid-identifier>')).toMatchSnapshot();
    });

    it('of numeric literal', () => {
        expect(new ENREName('Num', '123')).toMatchSnapshot();

        expect(ENREName.fromString('<Num 1e-10>')).toMatchSnapshot();
        // TODO: Add more tests according to the form of number
    });

    it('of class private identifier', () => {
        expect(new ENREName('Pvt', 'foo')).toMatchSnapshot();

        expect(ENREName.fromString('<Pvt foo>')).toMatchSnapshot();
    });

    it('of unknown', () => {
        expect(new ENREName('Unk')).toMatchSnapshot();

        expect(ENREName.fromString('<Unknown>')).toMatchSnapshot();
    });

    describe('calculated properties', () => {
        it('codeName', () => {
            expect(new ENREName('Pvt', 'foo').codeName).toEqual('#foo');
        });

        it('codeLength', () => {
            expect(new ENREName('Pvt', 'foo').codeLength).toEqual(4);
            expect(ENREName.fromString('<Anon Block>').codeLength).toEqual(0);
            expect(ENREName.fromString('<Block 1:1>').codeLength).toEqual(0);
        });

        it('printable', () => {
            expect(new ENREName('Str', 'a-not-valid-identifier').string).toEqual('<Str a-not-valid-identifier>');
        });
    });
});
