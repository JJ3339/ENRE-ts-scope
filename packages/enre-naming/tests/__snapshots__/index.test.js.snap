// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ENREName of anonymous ArrowFunction 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "ArrowFunction",
}
`;

exports[`ENREName of anonymous ArrowFunction 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "ArrowFunction",
}
`;

exports[`ENREName of anonymous Block 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Block",
}
`;

exports[`ENREName of anonymous Block 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Block",
}
`;

exports[`ENREName of anonymous Block 3`] = `
ENREName {
  "extra": "1:1",
  "kind": "Anon",
  "payload": "Block",
}
`;

exports[`ENREName of anonymous Block 4`] = `
ENREName {
  "extra": "1:1",
  "kind": "Anon",
  "payload": "Block",
}
`;

exports[`ENREName of anonymous Class 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Class",
}
`;

exports[`ENREName of anonymous Class 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Class",
}
`;

exports[`ENREName of anonymous Function 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Function",
}
`;

exports[`ENREName of anonymous Function 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Anon",
  "payload": "Function",
}
`;

exports[`ENREName of class private identifier 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Pvt",
  "payload": "foo",
}
`;

exports[`ENREName of class private identifier 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Pvt",
  "payload": "foo",
}
`;

exports[`ENREName of file name 1`] = `
ENREName {
  "extra": undefined,
  "kind": "File",
  "payload": "foo.js",
}
`;

exports[`ENREName of file name 2`] = `
ENREName {
  "extra": undefined,
  "kind": "File",
  "payload": "foo.js",
}
`;

exports[`ENREName of numeric literal 1`] = `
ENREName {
  "extra": 123,
  "kind": "Num",
  "payload": "123",
}
`;

exports[`ENREName of numeric literal 2`] = `
ENREName {
  "extra": 1e-10,
  "kind": "Num",
  "payload": "1e-10",
}
`;

exports[`ENREName of package 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Pkg",
  "payload": "foo",
}
`;

exports[`ENREName of package 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Pkg",
  "payload": "foo",
}
`;

exports[`ENREName of signature declaration Callable 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "Callable",
}
`;

exports[`ENREName of signature declaration Callable 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "Callable",
}
`;

exports[`ENREName of signature declaration NumberIndex 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "NumberIndex",
}
`;

exports[`ENREName of signature declaration NumberIndex 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "NumberIndex",
}
`;

exports[`ENREName of signature declaration StringIndex 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "StringIndex",
}
`;

exports[`ENREName of signature declaration StringIndex 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Sig",
  "payload": "StringIndex",
}
`;

exports[`ENREName of string literal 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Str",
  "payload": "a-not-valid-identifier",
}
`;

exports[`ENREName of string literal 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Str",
  "payload": "a-not-valid-identifier",
}
`;

exports[`ENREName of unknown 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Unk",
  "payload": undefined,
}
`;

exports[`ENREName of unknown 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Unk",
  "payload": undefined,
}
`;

exports[`ENREName of valid string identifier 1`] = `
ENREName {
  "extra": undefined,
  "kind": "Norm",
  "payload": "foo",
}
`;

exports[`ENREName of valid string identifier 2`] = `
ENREName {
  "extra": undefined,
  "kind": "Norm",
  "payload": "foo",
}
`;
